casepro/msgs/models.py

```diff
@@ -417,6 +429,15 @@ def get_history(self):
         """
         return self.actions.select_related('created_by', 'label').order_by('-pk')
 
+    def get_lock(self, user_id):
+        if self.locked_on and self.locked_by:
+            if self.locked_on > (now() - timedelta(seconds=MESSAGE_LOCK_SECONDS)):
+                if self.locked_by.id != user_id:
```

Comment:
Also use `locked_by_id` or ensure `locked_by` is prefetched to avoid db hits when calling this on a list of messages. Maybe include a `self.assertNumQueries` check in the tests to make sure we're not missing any prefetches.